initializr-version-resolver/src/main/java/io/spring/initializr/versionresolver/MavenResolverDependencyManagementVersionResolver.java

```diff
@@ -81,6 +82,15 @@ class MavenResolverDependencyManagementVersionResolver implements DependencyMana
 		LocalRepository localRepository = new LocalRepository(cacheLocation.toFile());
 		this.repositorySystem = serviceLocator.getService(RepositorySystem.class);
 		session.setLocalRepositoryManager(this.repositorySystem.newLocalRepositoryManager(session, localRepository));
+		final Properties systemProperties = new Properties();
```

Comment:
Can you explain this change and why you think it's necessary. I remember that something was off so I'd like to understand how you found out about this.

Reply:
Sure,

When we upgrade from maven provider 3.8.7 to 3.9.1, we get ModelBuildingException as you mentioned in https://github.com/spring-io/initializr/issues/1399.
This exception occurs at line 65 in [JdkVersionProfileActivator class](https://github.com/apache/maven/blob/maven-3.9.1/maven-model-builder/src/main/java/org/apache/maven/model/profile/activation/JdkVersionProfileActivator.java) when the value for
java.version is accessed from system properties.

In version 3.8.7, system properties value was already being set in [MavenRepositorySystemUtils 3.8.7](https://github.com/apache/maven/blob/maven-3.8.7/maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java) but were not being set in [MavenRepositorySystemUtils 3.9.1](https://github.com/apache/maven/blob/maven-3.9.1/maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java).

Thus I simply added the missing system properties in the way originally present in MavenRepositorySystemUtils 3.8.7.